primitives/src/v2/mod.rs

```diff
@@ -790,6 +809,12 @@ impl From<u32> for GroupIndex {
 	}
 }
 
+impl From<GroupIndex> for u32 {
+	fn from(i: GroupIndex) -> Self {
+		i.0
+	}
+}
+
```

Comment:
What is the motivation to add these? The reason we have typed indices is to ensure type-safety, that is we don't accidentally index a vector of groups with a validator index. With these conversions to u32, it's easy to make a mistake with `.into()`.

Reply:
Using another approach which should work better.